InstanceList: translate the component names

mlutfy opened the pull request:
Overview
----------------------------------------

Fixes the translation of component names in "Reports > All Reports".

Before
----------------------------------------

Incorrect use of `ts` to concatenate strings. Often translations didn't make sense, either because of an awkward ordering of the words, or because the component name it not correctly translated.

Ex: "Contact Reports" in English translates to "Contact Доклади" in Bulgarian, or "Contact Rapports" in French.

![image](https://github.com/user-attachments/assets/1ace5f31-ca43-4f8a-8ea7-3e5f75d9cbcf)

![image](https://github.com/user-attachments/assets/fccb18e6-61a8-4e99-bd7a-1c4faaf1ef20)

After
----------------------------------------

Translated component names, and I don't think it is necessary to repeat "Report" all the time.

![image](https://github.com/user-attachments/assets/c79d0455-b4bc-42c5-b46d-a2d28879292c)

Comments
----------------------------------------

I removed some CSS IDs and classes that were using the component names, because it becomes a bit unpredictable, and I don't see why anyone would need this.

The screen also shows `fr_CA`, where we chose to translate component names such as "CiviEvent" to just "Événements" (Events), because "CiviÉvénements" looks silly, and keeping it in English is not an option. I noticed some other languages just keep it in English (such as Bulgarian).

mlutfy committed:
Report InstanceList and TemplateList: translate the component names

And removed a font-size directive on tables.

ufundo commented:
Otherwise code looks good to me, thanks @mlutfy :+1:

colemanw commented:
Well, I hate the idea of using a translated string for array keys. That means faulty translations (like two components getting the same translation) could cause really weird stuff to happen in the application logic.
Couldn't we add a new array value instead?

mlutfy commented:
I fixed the key/translation, and also fix the Report Template list, which is somewhat similar.

- I removed the chevrons, because they can be confused with icons to toggle fieldsets
- Removed some hardcoded tiny font-sizes
- "Existing Report(s)" becomes "Existing Reports", easier to read

Before:

![image](https://github.com/user-attachments/assets/9aecdfd0-f1d6-4664-9ef7-e0cf24b6e31b)

After:

![image](https://github.com/user-attachments/assets/1915080a-d5d1-4d1b-a044-a7fc8896394f)

(and yes, poor life choices, to waste so much time on this)

colemanw commented:
Thanks @mlutfy. CiviCRM depends on people's poor choices 😆

seamuslee001 commented on CRM/Report/Page/TemplateList.php:
```diff
@@ -69,15 +70,26 @@ public static function &info($compID = NULL, $grouping = NULL) {
+        }
+        $rows[$component_name]['label'] = $label;
+      }
+      $rows[$component_name]['list'][$dao->value]['title'] = _ts($dao->label);
```
I know its the same as before effectively but what is `_ts` compared to `ts` ? @mlutfy

mlutfy replied:
According to `functions.php`, `_ts` is for invalid uses of `ts` that would normally throw linting errors (because we should not pass variables to the ts function).
